schema/schema.cc

```diff
@@ -1706,6 +1707,34 @@ schema_ptr schema_builder::build(compact_storage cp) {
     return with(cp).build();
 }
 
+internal_column_set::internal_column_set(bytes prefix_arg,
+    std::vector<schema::column> regular_columns_arg,
+    std::vector<schema::column> static_columns_arg)
+    : prefix(std::move(prefix_arg))
+    , regular_columns(std::move(regular_columns_arg))
+    , static_columns(std::move(static_columns_arg))
+{
+    if (!prefix.starts_with('$')) {
+        throw std::logic_error(fmt::format("Column set prefix \"{}\" must start with the '$' symbol",
+            to_string_view(prefix)));
+    }
+    const auto all_cols = std::views::join(std::array{
+        std::span{regular_columns}, 
+        std::span{static_columns}
+    });
+    if (all_cols.empty()) {
+        throw std::logic_error(
+            fmt::format("Column set \"{}\" must contain at least one column", to_string_view(prefix)));
+    }
+    for (const auto& c: all_cols) {
+        if (!c.name.starts_with(prefix)) {
+            throw std::logic_error(
+                fmt::format("Column name \"{}\" must start with the column set prefix \"{}\"",
+                    to_string_view(c.name), to_string_view(prefix)));
+        }
+    }
```

Comment:
I think those checks are a little excessive - this will be caught during review.

Better convert to on_internal_error, so if it does find something, we'll crash in QA rather than have to track it down.